src/java/org/apache/cassandra/index/Index.java

```diff
@@ -424,9 +424,10 @@ default RowFilter.CustomExpression customExpressionFor(TableMetadata metadata, B
      * @param restriction restriction that requires current index
      * @param columnIndex idx of the indexed column in returned row
      * @param options query options
+     * @param cqlRows list of lists of byte buffers based on the returned rows
      * @return comparator that for post-query ordering; or null if not supported
      */
-    default Comparator<List<ByteBuffer>> getPostQueryOrdering(Restriction restriction, int columnIndex, QueryOptions options)
+    default Comparator<List<ByteBuffer>> getPostQueryOrdering(Restriction restriction, int columnIndex, QueryOptions options, List<List<ByteBuffer>> cqlRows)
```

Comment:
it feels weird to pass the list of stuff you're going to sort, to the method that creates the comparator

Reply:
I’m unsure how to achieve this without passing the `cqlRows` into `getPostQueryOrdering`. How would you suggest approaching this? I explained why I passed `cqlRows` below.

This change was to avoid comparing the left and right vectors directly with the target following their deserialization but to enable to compare their scores directly. For this reason, the scores had to be extracted from a map of vector buffers and their scores. However, the byte buffers were needed beforehand for this to happen, thus it required to pass the result rows (list of list of byte buffers) via the `orderingComparator.prepareFor` to `StorageAttachedIndex.getPostQueryOrdering` to create the map there.